src/main/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentStepExecution.java

```diff
@@ -25,18 +27,6 @@
 
     private static final long serialVersionUID = 1L;
 
-    @StepContextParameter
-    private transient TaskListener listener;
-
-    @StepContextParameter
-    private transient Run<?, ?> build;
-
-    @StepContextParameter
-    private transient Launcher launcher;
-
-    @StepContextParameter
-    private transient FilePath workspace;
-
     @Inject(optional = true)
     private SSHAgentStep step;
```

Comment:
And get rid of this field too, and switch the supertype of `SSHAgentStep` from `AbstractStepImpl` to plain `Step`, so that its `start` method creates the `SSHAgentStepExecution` programmatically passing `this` (no Guice magic); the descriptor would also become plain `StepDescriptor`, so for `getRequiredContext` you use (say) `ImmutableSet.of` to list the context types you will need (`TaskListener` etc. as above). You should just be able to follow the deprecation warnings.

Reply:
I'm definitely not an expert, but is this a safe change, i.e. won't this lead to deserialization problems for builds that was ongoing during plugin upgrade?

Reply:
> won't this lead to deserialization problems for builds that was ongoing during plugin upgrade

Normally this is why you use the constructor of `AbstractStepExecutionImpl` that is _not_ deprecated. In this case it is true that the `Step` instance is included in the serial state as well (that is not typical), so yes this would need to retain the original deprecated `AbstractStepImpl` supertype, yet `start` is `final`…hmm. Maybe better to deprecate `SSHAgentStep`, leave it untouched, but move its descriptor to a new `Step` implementation, and have `SSHAgentStepExecution.readResolve` switch fields.

All this is going to be kind of complicated, so better to not do it in this PR which is not really related. Even just switching `@StepContextParameter`s to `getContext().get(…)` is not quite safe, as that affects the computed mandatory context for the deprecated `AbstractStepDescriptorImpl`. Can we rip the conversion stuff out of this PR and keep it focused on removing `Launcher` (and, optionally, `TaskListener`) from the remoted state?